packages/core/src/operations.ts

```diff
@@ -229,9 +230,18 @@ export class TxOperations implements Omit<Client, 'notify'> {
     const hierarchy = this.client.getHierarchy()
     const mixClass = Hierarchy.mixinOrClass(doc)
     if (hierarchy.isMixin(mixClass)) {
-      // TODO: Rework it is wrong, we need to split values to mixin update and original document update if mixed.
+      const { docUpdate, mixinUpdate } = this.splitMixinUpdate(update, mixClass)
+
       const baseClass = hierarchy.getBaseClass(doc._class)
-      return this.updateMixin(doc._id, baseClass, doc.space, mixClass, update, modifiedOn, modifiedBy)
+
+      const result: Promise<TxResult>[] = []
+      if (Object.keys(docUpdate).length > 0) {
+        result.push(this.updateDoc(baseClass, doc.space, doc._id, docUpdate, retrieve, modifiedOn, modifiedBy))
+      }
+      if (Object.keys(mixinUpdate).length > 0) {
```

Comment:
also please expect, it could be multiple mixins in one update.

Reply:
Fixed